src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java

```diff
@@ -1933,6 +1933,140 @@ public Course addCourseWithThreeFileUploadExercise() {
+                    modelSubmissionService.save(submission, modelingExercise, "student" + j);
+                    studentParticipationRepo.save(participation);
+                    if (numberOfAssessments >= j) {
+                        Result result = new Result();
+                        result = resultRepo.save(result);
+                        result.setSubmission(submission);
+                        result.completionDate(pastTimestamp);
+                        result.resultString("1 of 10 points");
+                        result.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+                        result.setAssessor(currentUser);
+                        submission.addResult(result);
+                        participation.addResult(result);
+                        studentParticipationRepo.save(participation);
+                        modelingSubmissionRepo.save(submission);
+                        resultRepo.save(result);
+                        if (numberOfComplaints >= j) {
+                            Complaint complaint = typeComplaint ? new Complaint().complaintType(ComplaintType.COMPLAINT)
+                                    : new Complaint().complaintType(ComplaintType.MORE_FEEDBACK);
+                            complaint.setResult(result);
+                            complaint = complaintRepo.save(complaint);
+                            if (numberComplaintResponses >= j) {
+                                ComplaintResponse complaintResponse = createInitialEmptyResponse(typeComplaint ? "tutor5" : currentUser.getLogin(), complaint);
+                                complaintResponse.getComplaint().setAccepted(true);
+                                complaintResponse.setResponseText(typeComplaint ? "Accepted" : "SomeMoreFeedback");
+                                complaintResponseRepo.save(complaintResponse);
+                                complaint.setComplaintResponse(complaintResponse);
+                                complaintRepo.save(complaint);
+                            }
+                        }
+                    }
+                }
+
+            }
+            else if ((i % 3) == 1) {
+                TextExercise textExercise = ModelFactory.generateTextExercise(pastTimestamp, pastTimestamp, futureTimestamp, course);
+                textExercise.setTitle("Text" + i);
+                course.addExercises(textExercise);
+                course = courseRepo.save(course);
+                exerciseRepo.save(textExercise);
+                for (int j = 1; j <= numberOfSubmissionPerExercise; j++) {
+                    TextSubmission submission = ModelFactory.generateTextSubmission("submissionText", Language.ENGLISH, true);
+                    submission = saveTextSubmission(textExercise, submission, "student" + j);
+                    if (numberOfAssessments >= j) {
+                        Result result = new Result();
```

Comment:
This whole section here seems identical in all the else cases. Maybe you could extract it into a new method?

Reply:
I refactored it a bit to make it more compact